beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -4671,18 +4679,20 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
         // If the current slot is already equal to the proposal slot (or we are in the tail end of
         // the prior slot), then check the actual weight of the head against the re-org threshold.
         let head_weak = if fork_choice_slot == re_org_block_slot {
-            info.head_node.weight < info.re_org_weight_threshold
+            info.head_node.weight < info.re_org_head_weight_threshold
         } else {
             true
         };
         if !head_weak {
             return Err(DoNotReOrg::HeadNotWeak {
                 head_weight: info.head_node.weight,
-                re_org_weight_threshold: info.re_org_weight_threshold,
+                re_org_head_weight_threshold: info.re_org_head_weight_threshold,
             }
             .into());
         }
 
+        // TODO(is_parent_strong) do we need an is_parent_strong check here
```

Comment:
To follow the spec, we should check it only in the case where `fork_choice_slot == re_org_block_slot`. I.e. when we are proposing at slot `N` and running fork choice for slot `N`. By this time the attestations from slot `N - 1` have been applied, so we expect to see ~2 slots worth of attestations reinforcing the parent block. In Lighthouse we evaluate fork choice for slot `N` around 500ms before the slot starts, so this gives us time to "give up" on an attempted re-org and stop suppressing fcU _if_ we see that the re-org is likely to fail (`should_override_forkchoice_update(..) == False`).

The relevant part of the spec is:

```python
    # Check the head weight only if the attestations from the head slot have already been applied.
    # Implementations may want to do this in different ways, e.g. by advancing
    # `store.time` early, or by counting queued attestations during the head block's slot.
    if current_slot > head_block.slot:
        head_weak = is_head_weak(store, head_root)
        parent_strong = is_parent_strong(store, parent_root)
    else:
        head_weak = True
        parent_strong = True
```

From https://github.com/ethereum/consensus-specs/blob/dev/specs/bellatrix/fork-choice.md#notify_forkchoice_updated